models/Asset.php

```diff
@@ -1594,10 +1581,9 @@ public function getMetadata($name = null, $language = null, $strictMatch = false
         if (is_array($metaData)) {
             foreach ($metaData as $md) {
                 $md = (array)$md;
-                if (!$raw) {
-                    $md['data'] = $convert($md);
+                if ((empty($md['language']) && !$strictMatch) || ($md['language'] == $language)) {
+                    $result[] = $raw ? $md : $convert($md);
```

Comment:
also like above

Reply:
@kingjia90 I have updated accordingly and now the tests are happy :)